FPGA_Open_Speech_Tools/README.md

```diff
@@ -0,0 +1,316 @@
+ INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+ PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+ FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+ Ross K. Snider
+ Justin P. Williams
+ E. Bailey Galacci
+ Flat Earth Inc
+ 985 Technology Blvd
+ Bozeman, MT 59718
+ support@flatearthinc.com
+
+This README is focused on explaining how to use each of the blocks provided within this library.
+There will also be detailed explanation on the design of each block.
+
+The DSP FPGA Accelerated Toolbox is designed as a hardware friendly support to the existing hdllib library. 
+
+## Table of Contents
+1. Programmable Look-Up Table
+2. Programmable Look-Up Table, Linear Scale
+3. Static Upclocked FIR 
+4. Programmable Upclocked FIR 
+5. Circular Buffer, Variable Delay 
+
+## Programmable Look-Up Table (PLUT)
+A log-scale lookup table, designed to emulate transcendental functions with minimal resource usage.  
+Accuracy is boosted through the use of linear interpolation, without the use of division.  On initialization, the PLUT represents a single user-defined function with input "xIn". xIn does not need to be defined in the workspace,  as it will be defined in the initialization script of the block in the local workspace. The PLUT adjusts its own size to fit the user-defined accuracy. Note points are pseudo-logarithmically spaced to increase precision throughout the input space.
+
+Tables can be initialized on system startup and rewritten during runtime using the Table_Wr lines.
+Min input cannot be 0, because inputs are pseudo-logarithmically spaced.
+
+Input and Output set can be saved as "xIn" and "tableInit" within the file "<path name> init.mat" for ease of reprogramming. Input values remain constant until the mask parameters are updated. Table Output values can be reprogrammed during runtime using the Wr_Data, Wr_En and RW_Addr lines.
+
+Uses pseudo-logarithmic spacing of inputs for maximum accuracy in non-linear functions, as well as linear interpolation between points to further increase accuracy. Because of this, it is assumed that function input is unsigned. If this is undesired, a second table would need to be implemented to cover negative values, or some other solution outside the block.
+
+Capable of automatically determining required table size to meet a given accuracy requirement within the entire data range, or can be manually set to a specific size with M_bits and N_bits. N_bits determines the data range as a power of 1/2 of the maximum input. M_bits determines the number of bits in the address used to increase function precision, by increasing the number of points stored between each log2 spaced input point.
+
+### Mask Parameters 
```

Comment:
I wonder if having the input/output port descriptions before the mask parameters would be better.